### PR TITLE
[JSON-API/trigger-service] Refactor db conn

### PR DESCRIPTION
**Pull request by akshayshirahatti-da**
Initial work for sharing Connection and JdbcConfig between json_api and trigger service, is purely refactoring.

- Move ExceptionOps from ledger-service/utils to //libs-scala/scala-utils
- extract connection and JdbcConfig from //ledger-service to independent //libs-scala/db-utils module.

Changelog_begin
Changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@ package com.daml.http
 
 import com.daml.grpc.adapter.ExecutionSequencerFactory
 import com.daml.ledger.client.configuration.LedgerClientConfiguration
-import com.daml.util.ExceptionOps._
+import com.daml.scalautil.ExceptionOps._
```
Comment (victormueller-da) [inline ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala:8]: I think `scalautil` is a bit weird, not my best idea yet but I think `scala.util` sounds better.

File: ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala [hunk 2]
```diff
@@ -25,7 +26,6 @@ import scalaz.std.scalaFuture._
 import scalaz.std.string._
 import scalaz.syntax.tag._
 import scalaz.{-\/, EitherT, \/, \/-}
-
 import Config.QueryStoreIndex
```
Comment (victormueller-da) [inline ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala:29]: If possible, make sure that in the end the whitespace changes are minimal (If you do that already, sorry for calling you out here. I haven't reviewed your code before).

File: ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala (hunk 1 above)
Comment (victormueller-da) [reply ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala:8]: ups, wrong account. I'm here usually only using @realvictorprm ;)

**Comment by realvictorprm** [comment]
As you asked for some review. I can tell that I think this is going into the right direction. I'll take another look as soon as you're marking this ready to be reviewed (so no-draft anymore).

**Comment by realvictorprm** [comment]
Side note, if you want to do _me_ a favour, add the component name to the PR too, both label and title. If you wonder why, labels are useful but it's still faster to see for newest PR's if it's the json api because a bold prefix can be seen faster. I usually use `[JSON-API]`.

However it's not necessary and I guess not everyone would agree with that, however I'm always in favour of more structure :)

File: ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala (hunk 2 above)
Comment (akshayshirahatti-da) [reply ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala:29]: cheers will look at cleaning this up

File: ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala (hunk 1 above)
Comment (akshayshirahatti-da) [reply ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala:8]: While I agree, I just moved it to an existing package/module which I thought was most appropriate for this to live at i.e `libs-scala\scala-utils`.

Comment (realvictorprm) [reply ledger-service/http-json-ledger-client/src/main/scala/com/daml/http/LedgerClientBase.scala:8]: Oh I see... Geez, nothing we can do about then :(

Thanks anyways :)

File: libs-scala/db-utils/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    scala_deps = [
+        "@maven//:org_scalatest_scalatest",
+        "@maven//:org_scalaz_scalaz_core",
+        "@maven//:org_tpolecat_doobie_core",
+        "@maven//:org_typelevel_cats_core",
+        "@maven//:org_typelevel_cats_effect",
+        "@maven//:org_typelevel_cats_kernel",
+    ],
+    scalacopts = lf_scalacopts,
+    # data = ["//docs:quickstart-model.dar"],
```
Comment (victormueller-da) [inline libs-scala/db-utils/BUILD.bazel:59]: I think this can be removed

**Review by victormueller-da — APPROVED** [review]
Overall looks good to me

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by akshayshirahatti-da** [comment]
work done as part of [#10155](https://github.com/digital-asset/daml/issues/10155)

**Pull request merged**